### ISSUE TITLE
DS session & multiple components

### ISSUE THREAD

**Issue by ybznek** [OpenSCAP/openscap#545]
```
$ oscap xccdf eval  ./r3010-datastream.xml 
...
Title   Test the Applicability processing capabilities
Rule    xccdf_gov.nist_rule_r3010-0_6
Result  notchecked

OpenSCAP Error: File r3010-cpe-oval has already been register with Source DataStream session: ./r3010-datastream.xml [ds_sds_session.c:239]
```

[content.zip - NIST certification files](https://github.com/OpenSCAP/openscap/files/500590/archive.zip)

**Issue:**
CPE component is "loaded" twice and oscap function `ds_sds_session_register_component_source()` returns -1 for such operation. The reason why it still work is that we don't check return value of the function for error.

Currently it seems to be working, but cause error message even for successful scan and it cause memory leak.

**Comment by ybznek** [comment]
After merge of https://github.com/OpenSCAP/openscap/pull/534 I could provide more details.

**Comment by ybznek** [comment]
@jan-cerny What do you think, is it certification blocker? Functionality is OK, but with such error.

**Comment by ybznek** [comment]
We were ignoring return value of ds_sds_dump_local_component(). It caused that if occured some error with "double registration" of component, we ignored it. Maybe this error isn't too fatal, but it could hide other errors. 

So I started to check return value of ```ds_sds_dump_local_component()```. But it broke already working use case. So I started to ignore return value of ```ds_sds_session_register_component_source()``` to make it working again.

TLDR:
https://github.com/OpenSCAP/openscap/blob/maint-1.2/src/DS/sds.c#L242
Replace `return 0` by `return ds_sds_session_register_component_source(session, relative_filepath, component_source);` & make it working.

**Comment by jan-cerny** [comment]
@ybznek @mpreisler This is not a certification blocker, it is just a hidden mistake in our code that doesn't manifest itself.

To be more clear, with current upstream, there is no error message on stderr. Also no other failure appears. The testcase outlined above passes.

The only problem is that we attempt to register one of the datastream components in our internal model twice, which is over-abundant. Second attempt is not successful. Currently we ignore it, but we should fix in a way that all the components will be registered only once.

**Comment by ybznek** [comment]
Eh, You are right.
But it is still in verbose mode log. I don't know why it "moved" from stderr.
```
$ myoscap xccdf eval --verbose ERROR --verbose-log-file /dev/stderr ./r3010-datastream.xml
E: oscap: (ds_sds_session.c:257:ds_sds_session_register_component_source()) File r3010-cpe-oval has already been register with Source DataStream session: ./r3010-datastream.xml
```

**Comment by jan-cerny** [comment]
@ybznek Could it be because we change buffering in the reporting callback while downloading remote resources?
